Assets/MixedRealityToolkit.SDK/Features/Utilities/InputUtils.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.WindowsDevicePortal;
+using System.Runtime.InteropServices.WindowsRuntime;
+using UnityEngine;
+using UnityEngine.Experimental.XR;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Utilities for accessing input state.
+    /// </summary>
+    public static class InputUtils
+    {
+        /// <summary>
+        /// Gets the ray representing the position and direction of the user's head.
+        /// </summary>
+        /// <returns>The ray the head gaze</returns>
+        public static Ray GetHeadGazeRay()
+        {
+            return new Ray(CameraCache.Main.transform.position, CameraCache.Main.transform.forward);
```

Comment:
I don't know if this really fits in this PR. It just overall feels odd that we have pointers and input sources and a gaze provider and an eye gaze provider. Like pointers have their own `Rays` element. Seems like things are different but a lot of overlap

Reply:
Great observations, I agree it's out of the scope of this PR. Let's work together to improve this :-)

Reply:
Maybe a concrete thing we can do is to rename the pointers Rays property to Pointer.Segments or Pointer.RaycastSegments.